### ISSUE TITLE
Invalid salt revision when performing checkpw

### ISSUE THREAD

**Issue by devxpy** [Erly/dbcrypt#2]
Flutter 0.5.1

```
new DBCrypt().checkpw('test', '\$2b\$12\$0O5iZrh9JNnOgBP/NprFBe2SS5scgrLsA.Dx6DsmhL3VLQpN/q4Uy')
```
results in following error
```
E/flutter (13825): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter (13825): Invalid salt version
E/flutter (13825): #0      DBCrypt.hashpw (package:dbcrypt/dbcrypt.dart:376:7)
E/flutter (13825): #1      DBCrypt.checkpw (package:dbcrypt/dbcrypt.dart:474:30)
E/flutter (13825): #2      check_secret_key (file:///home/dev/AndroidStudioProjects/secure_pdf_viewer/lib/src/util.dart:20:20)
E/flutter (13825): <asynchronous suspension>
E/flutter (13825): #3      LoginFormState.build.<anonymous closure> (file:///home/dev/AndroidStudioProjects/secure_pdf_viewer/lib/main.dart:60:18)
E/flutter (13825): <asynchronous suspension>
E/flutter (13825): #4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:494:14)
E/flutter (13825): #5      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:549:30)
E/flutter (13825): #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102:24)
E/flutter (13825): #7      TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:161:9)
E/flutter (13825): #8      TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:94:7)
E/flutter (13825): #9      PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:315:9)
E/flutter (13825): #10     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
E/flutter (13825): #11     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
E/flutter (13825): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:143:19)
E/flutter (13825): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:121:22)
E/flutter (13825): #14     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:101:7)
E/flutter (13825): #15     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:64:7)
E/flutter (13825): #16     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:48:7)
E/flutter (13825): #17     _invoke1 (dart:ui/hooks.dart:134:13)
E/flutter (13825): #18     _dispatchPointerDataPacket (dart:ui/hooks.dart:91:5)
```

**Comment by devxpy** [comment]
The generated hash was from python, which has the $2b$ prefix by default.

This lib seesm to work on $2a$ prefix

After changing that, it seems to work fine !

**Comment by Erly** [comment]
The wikipedia BCrypt article says that $2b$ is a new version of BCrypt released in 2014 that fixes a bug in the OpenBSD implementation of BCrypt. So I'm gonna reopen the issue since this should be supported, and I'll check if this library has that same bug.

**Comment by Erly** [comment]
Fixed on 693183ecb57fca0d9cc068f2029109e0c6b17c1a should be released this week with 1.0.0 version
